shared/public/CollisionGrid.h

```diff
@@ -98,43 +99,53 @@ class CollisionGrid {
                 }
             }
         }
+
+        bool colliding = false;
         for (int16_t y = indexRange.yMin; y <= indexRange.yMax; y++) {
             for (int16_t x = indexRange.xMin; x <= indexRange.xMax; x++) {
                     for (const auto &rect : gridRects[y][x]) {
                         if (CollisionUtil::checkRectCollision(projectedRectangle, rect)) {
-                            return 1;
+                            if (alwaysInsert) {
+                                colliding = true;
+                                break;
+                            } else {
+                                return 1;
+                            }
                         }
                     }
                     for (const auto &circle : gridCircles[y][x]) {
                         if (CollisionUtil::checkRectCircleCollision(projectedRectangle, circle)) {
-                            return 1;
+                            if (alwaysInsert) {
+                                colliding = true;
+                                break;
+                            } else {
+                                return 1;
+                            }
                         }
                     }
-
             }
         }
 
-        // Only insert, when not colliding
         for (int16_t y = indexRange.yMin; y <= indexRange.yMax; y++) {
             for (int16_t x = indexRange.xMin; x <= indexRange.xMax; x++) {
```

Comment:
hier iterierst du nochmal durchs ganze Gitter durch das du vorhin schon iteriert bist. Kannst das push_back einfach hochziehen in die andere Loop

Reply:
Das Problem ist, dass wir im Standard-Fall erst sicher sein müssen, dass wir nicht colliden, da wir sonst das Label ja gar nicht einfügen wollen.
Möglich wäre die Optimierung, dass wir die Insertion beim alwaysInsert oben machen. Dann haben wir einfach zwei Stellen dafür.